File: practice.py
class Atm:
    def __init__(self):
        self.pin = ""
        self.balance = 0
        
    def menu(self):
        user_input = input("""
                         Hi! how can I help you?
                         1. Create account
                         2. deposite money
                         3. Change pin
                         4. Withdraw balance
                         5. Exit
                         """)
        if user_input == "1":
            self.create_pin()
        elif user_input == "2":
            self.deposite_money()
        elif user_input == "3":
            self.change_pin()
        elif user_input == "4":
            self.withdraw()
        else:
            exit()
        
    def create_pin(self):
        user_pin = input("Enter your pin: ")
        self.pin = user_pin
        print("Your pin was created successfully")
        self.menu()
        
    def deposite_money(self):
        user_pin=input("enter your pin: ")
        if user_pin==self.pin:
            user_balance=input("enter money you want to deposite : ")
            self.balance=user_balance
            self.menu()
        else:
            print("enter valid pin!")
            self.menu()
        
    


if __name__ == "__main__":
    obj = Atm()
    obj.menu()  # Check syntax error fixed

File: practice.cpp
#include <bits/stdc++.h>
using namespace std;

class Atm
{
private:
    string pin;
    int balance;

public:
    // Constructor to initialize pin and balance
    Atm(string pin = "", int balance = 0) : pin(pin), balance(balance) {}

    // Function to display the menu
    void menu()
    {
        cout << "\nATM Menu:\n";
        cout << "1. Create Account\n";
        cout << "2. Deposit Money\n";
        cout << "3. Change PIN\n";
        cout << "4. Withdraw Money\n";
        cout << "5. Exit\n";
        cout << "Enter your choice: ";
    }

    // Function to create an account (set PIN)
    void create_account()
    {
        cout << "Enter your new PIN: ";
        cin >> pin;
        cout << "Account created successfully!\n";
    }
};

int main()
{
    // Create an instance of the Atm class
    Atm obj;

    int choice;
    while (true)
    {
        obj.menu(); // Display the menu
        cin >> choice;

        switch (choice)
        {
        case 1:
            obj.create_account();
            break;
        case 5:
            cout << "Exiting the program. Thank you!\n";
            return 0;

        default:
            cout << "Invalid choice! Please try again.\n";
        }
    }

    return 0;
}
